internal/olm/operator/uninstall.go

```diff
@@ -82,88 +89,116 @@ func (u *Uninstall) Run(ctx context.Context) error {
 		}
 	}
 
-	catsrc := &v1alpha1.CatalogSource{}
+	catsrcKey := client.ObjectKeyFromObject(catsrc)
 	if sub != nil {
-		catsrcKey := types.NamespacedName{
+		subObj = sub
+		// Use the subscription's catalog source data only if available.
+		keyFromSpec := types.NamespacedName{
 			Namespace: sub.Spec.CatalogSourceNamespace,
 			Name:      sub.Spec.CatalogSource,
 		}
-		if err := u.config.Client.Get(ctx, catsrcKey, catsrc); err != nil {
-			return fmt.Errorf("get catalog source: %v", err)
+		if catsrcKey.Name != "" && catsrcKey.Namespace != "" {
+			catsrcKey = keyFromSpec
 		}
 
-		csv, err := u.getInstalledCSV(ctx, sub)
-		if err != nil {
-			return fmt.Errorf("get installed CSV %q: %v", sub.Status.InstalledCSV, err)
+		// CSV name may either be the installed or current name in a subscription's status,
+		// depending on installation state.
```

Comment:
Just a quick question, what would be the `currentCSV` and `InstalledCSV` in the upgrade process. What is the condition when installed and currentCSV are different.

Reply:
From the [godocs](https://github.com/operator-framework/api/blob/master/pkg/operators/v1alpha1/subscription_types.go#L168-L174): installing (currently processing) vs. installed.